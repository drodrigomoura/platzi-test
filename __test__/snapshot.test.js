import { getCharacter } from '../snapshot'
import rick from '../rick.json'

describe("It's snapshot time", () => {
  it('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot()
  });

  it('Always fails the snapshot', () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20)
    }

    expect(user).toMatchSnapshot()
  });

  it('Exception for some values', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Rick Sanchez"
    }

    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    })
  });

})
