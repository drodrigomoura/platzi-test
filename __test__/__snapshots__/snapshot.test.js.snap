// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It's snapshot time Always fails the snapshot 1`] = `
Object {
  "createAt": 2021-07-23T02:01:14.872Z,
  "id": 15,
}
`;

exports[`It's snapshot time Exception for some values 1`] = `
Object {
  "id": Any<Number>,
  "name": "Rick Sanchez",
}
`;

exports[`It's snapshot time Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;
